src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1839,10 +1839,21 @@ private void SpillEvalStack()
 
                 // Declarations cannot have control flow, so we don't need to spill them.
                 if (operationOpt.Kind != OperationKind.FlowCaptureReference
-                    && operationOpt.Kind != OperationKind.DeclarationExpression
+                    && (operationOpt.Kind != OperationKind.DeclarationExpression || spillDeclarationExpressions)
                     && operationOpt.Kind != OperationKind.Discard
                     && operationOpt.Kind != OperationKind.OmittedArgument)
                 {
+                    if (operationOpt is IDeclarationExpressionOperation declarationExpression)
+                    {
+                        Debug.Assert(spillDeclarationExpressions);
+                        // We don't leave regular declarations in the CFG graph, so what we really want to capture is the
+                        // node underlying the declaration. Currently, this should only happen for `out` variable declarations,
+                        // which can only be field or local references. If we ever encounter a scenario where tuples get here,
+                        // we will need to update the logic to ensure we aren't erasing information.
+                        operationOpt = OperationCloner.CloneOperation(declarationExpression.Expression);
```

Comment:
>CloneOperation [](http://example.com/codeflow?start=55&length=14)

Is the clone necessary?

Reply:
Yes. The expression already has a `Parent` and we can't reset it.